lib/src/components/NxScrollRender/NxScrollRender.tsx

```diff
@@ -0,0 +1,137 @@
+        ref: leadingSpacerRefFn,
+        role: 'presentation',
+        className: 'nx-scroll-render__spacer',
+        style: { height: leadingSpacerHeight }
+      })}
+      {renderedRealChildren}
+      {React.createElement(spacerEl ?? 'div', {
+        ref: trailingSpacerRefFn,
+        role: 'presentation',
+        className: 'nx-scroll-render__spacer',
+        style: { height: trailingSpacerHeight }
+      })}
+    </>
+  ) : null;
+
+  const updateRendering = useThrottleCallback(useCallback(function updateRendering() {
+    const scrollTop = parentRef.current?.scrollTop ?? 0,
+        parentTop = parentRef.current?.getBoundingClientRect().top,
+        leadingSpacerTop = leadingSpacerRef.current?.getBoundingClientRect().top,
+        topDifference = (parentTop ?? 0) - (leadingSpacerTop ?? 0),
+        topTooClose = topDifference < normalizedChildHeight,
+        topTooFar = topDifference > normalizedChildHeight * 2,
+        adjust =
+          topTooClose ? dec :
+          topTooFar ? inc :
+          identity,
+        sumSpacerHeight = sumChildHeight - renderedChildHeight,
+        clampFirstRenderedChildIdx =
+            renderedChildCount == null ? always(0) : clamp(0, childCount - renderedChildCount),
+        clampSpacerHeight = clamp(0, sumSpacerHeight),
+        newFirstRenderedChildIdx = clampFirstRenderedChildIdx(
+            adjust(Math.floor(divOrZero(scrollTop, normalizedChildHeight)) - 2)
+        ),
+        newLeadingSpacerHeight = clampSpacerHeight(newFirstRenderedChildIdx * normalizedChildHeight),
+        newTrailingSpacerHeight = renderedChildCount == null ? 0 : clampSpacerHeight(
+            sumChildHeight - (newLeadingSpacerHeight + renderedChildCount * normalizedChildHeight)
+        );
+
+    setFirstRenderedChildIdx(newFirstRenderedChildIdx);
+    setLeadingSpacerHeight(newLeadingSpacerHeight);
+    setTrailingSpacerHeight(newTrailingSpacerHeight);
+  }, [childCount, normalizedChildHeight, renderedChildCount, renderedChildHeight, sumChildHeight]), 15);
+
+  const adjustedParent = React.cloneElement(fullParent, {
+    ref: parentRef,
+    children: renderedChildren,
+    onScroll: updateRendering
+  });
+
+  useLayoutEffect(function() {
+    if (parentRef.current) {
+      // note: not counting the spacer
+      const firstChild = parentRef.current.children[1],
+          secondChild = parentRef.current.children[2];
+
+      if (!secondChild) {
+        setChildHeight(0);
+      }
+      else {
+        const firstChildTop = firstChild?.getBoundingClientRect()?.top ?? 0,
+            secondChildTop = secondChild?.getBoundingClientRect()?.top ?? 0;
+
+        setChildHeight(secondChildTop - firstChildTop);
+      }
+
+      setParentHeight(parentRef.current.clientHeight);
+    }
+  }, []);
+
+  useResizeObserver(parentRef, useCallback(function({ contentBoxSize: [{ blockSize }] }) {
+    setParentHeight(blockSize);
+  }, []));
+
+  useEffect(updateRendering, [updateRendering]);
```

Comment:
Just wondering do we need the `useEffect` here? Seems like the `updateRendering` is triggered when scrolling on the parent.

Reply:
> Just wondering do we need the useEffect here? Seems like the updateRendering is triggered when scrolling on the parent.

It's needs to run not just during scrolling, but also on initial render in order to get the spacers sized correctly. If I comment out this line, the example starts with the scrollbar the wrong size.

Reply:
Actually, I've just realized that the spacer heights are derived entirely from other pieces of state and props, and so do not need to be state themselves; they can just be local variables.  Implementing that change has removed the need to call `updateRendering` on first render. 01a02340d

Reply:
> It's needs to run not just during scrolling, but also on initial render in order to get the spacers sized correctly. If I comment out this line, the example starts with the scrollbar the wrong size.

I see, that was for the initial render purpose.